enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,348 @@
+to make update recommendations especially in the case of upgrading from a homogeneous to a heterogeneous payload:
+
+- Have the heterogeneous graph include the homogeneous releases so CVO can make recommendations based on the current homogeneous release image.
+This is complicated by the fact that there will be multiple releases with the same version but different payloads. The trigger to update to a
+heterogeneous payload would be an option provided by the administrator through either the ClusterVersion or Infrastructure API to "move" to the
+heterogeneous graph.
+- Have both the graphs be completely independent of each other and have an option in one of the aforementioned APIs to make the cluster heterogeneous
+and CVO would request the heterogeneous graph, find the current release version and update to it.
+- Have an independent channel for heterogeneous payloads and have administrators switch channels if they want to move between homogeneous and
+heterogeneous payloads
+
+These options are under discussion and a finalized proposal will be put forth in the ticket for upgrades referenced above along with the API that would
+change to include the option to indicate a move to the heterogeneous payload.
+
+Cincinnati needs to [support][OCPPLAN-7643] parsing manifestlist images to be able to upgrade to a heterogeneous architecture payload.
+
+### Imagestreams
+
+Imagestreams need to better support manifestlist images. As of today they do recognize manifestlists, but it only imports images matching the control
+plane architecture. The idea here would be to import the manifest listed image in its entirety. This also has other implications such as the fact that
+the internal registry does not support manifest listed images. So, with full manifestlist support in imagestreams, pullthrough is not possible for
+manifestlists. We would ask users to use quay.io rather than use the internal registry if full manifestlist support is needed.
+
+The oc commands like debug, must-gather etc use the imagestreams and therefore would use an x86 image to deploy on the worker nodes which would fail
+with an ARM worker. As a workaround until imagestreams support importing all the architectures, they can be provided the manifest listed image with the
+`--image` parameter which would allow the commands to work on the worker nodes as well.
+
+### OLM
+
+Operators need to specify which architectures they support based on annotations. There also needs to be a mechanism to ensure that the architecture is
+actually supported if that label is present. Also, all architectures should share the same content.
+
+OLM would need to make sure that it deploys operators only to nodes whose architecture matches the architectures on which the operator is supported.
+This would mean that the OLM would have to be told which architectures an operator supports by the Console operator. Today, OLM has a subscription
```

Comment:
technically olm ought to know which architectures the operator supports.  but the admin installing the operator may not want to deploy it on all of those architectures.

so the olm piece of this is just updating the subscription api so that nodeaffinity can be specified.

the console piece will be giving the user a way to specify what that nodeaffinity should be, as part of the install (and defaulting it appropriately)

Reply:
updated.